holoviews/plotting/bokeh/raster.py

```diff
@@ -300,7 +299,28 @@ def get_data(self, element, ranges, style):
         return (data, mapping, style)
 
 
-class RGBPlot(ServerHoverMixin, LegendPlot):
+class SyntheticLegendMixin(LegendPlot):
+
+    def _init_glyphs(self, plot, element, ranges, source):
+        super()._init_glyphs(plot, element, ranges, source)
+        if not ('holoviews.operation.datashader' in sys.modules and self.show_legend):
+            return
+        try:
+            style = self.lookup_options(element, 'style')
+            legend = categorical_legend(element, backend=self.backend, cmap=style.options.get('cmap'))
```

Comment:
Should we add a test to verify that `cmap` is passed over?

Reply:
Done in [a37d653](https://github.com/holoviz/holoviews/pull/6662/commits/a37d653d76fa6095c428e5195d7fc6864d53fec8)